php-classes/ActiveRecord.class.php

```diff
@@ -2177,10 +2177,20 @@ protected function _setFieldValue($field, $value)
 
         if ($forceDirty || !array_key_exists($columnName, $this->_record) || $this->_record[$columnName] !== $value) {
             if (array_key_exists($columnName, $this->_record)) {
-                $this->_originalValues[$field] = $originalValue;
+                // only store original value if it hasn't been previously set
+                if (!array_key_exists($field, $this->_originalValues)) {
+                    $this->_originalValues[$field] = $originalValue;
+                } elseif ($this->_originalValues[$field] == $value) { // the value was reverted back to it's original value
+                    unset($this->_originalValues[$field]);
+                }
             }
             $this->_record[$columnName] = $value;
-            $this->_isDirty = true;
+            // set record dirty if there are still dirty fields remaining
```

Comment:
@themightychris I thought of that, but then I saw that there was a `$forceDirty` variable that was sometimes used. If we move the logic to the `_isDirty` expression, it'll prevent one from overriding/forcing a record's dirty-ness